core/src/test/java/com/linecorp/armeria/server/PathMappingTest.java

```diff
@@ -67,6 +72,26 @@ public void successfulOf() {
         assertThat(m.regex().get()).isEqualTo("^/files/(?<filePath>.*)$");
     }
 
+    @Test
+    public void withHttpHeaderInfo() {
+        final PathMapping m = PathMapping.of("/foo");
+
+        final PathMapping pathMappingWithInfo =
+                m.withHttpHeaderInfo(ImmutableSet.of(HttpMethod.GET), ImmutableList.of(), ImmutableList.of());
+        assertThat(pathMappingWithInfo.supportedMethods()).containsExactly(HttpMethod.GET);
+
+        assertThatThrownBy(
+                () -> pathMappingWithInfo.withHttpHeaderInfo(ImmutableSet.of(HttpMethod.GET),
+                                                             ImmutableList.of(), ImmutableList.of()))
+                .isInstanceOf(IllegalArgumentException.class);
+
+        assertThatThrownBy(
+                () -> m.withHttpHeaderInfo(ImmutableSet.of(),
+                                                             ImmutableList.of(), ImmutableList.of()))
```

Comment:
nit: could be merged into a single line.

Reply:
Oops!